agent/structs/config_entry_gateways_test.go

```diff
@@ -185,3 +268,87 @@ func TestIngressConfigEntry_Validate(t *testing.T) {
 		})
 	}
 }
+
+func TestIngressConfigEntry_ContainsService(t *testing.T) {
+	t.Parallel()
+
+	basicEntry := IngressGatewayConfigEntry{
+		Kind: "ingress-gateway",
+		Name: "ingress-web",
+		Listeners: []IngressListener{
+			{
+				Port:     1111,
+				Protocol: "http",
+				Services: []IngressService{
+					{
+						Name: "web",
+					},
+				},
+			},
+			{
+				Port:     1112,
+				Protocol: "tcp",
+				Services: []IngressService{
+					{
+						Name: "db",
+					},
+				},
+			},
+		},
+	}
+
+	cases := []struct {
+		name      string
+		service   ServiceID
+		entry     IngressGatewayConfigEntry
+		assertion func(require.TestingT, bool, ...interface{})
+	}{
+		{
+			name:      "exact name match",
+			service:   NewServiceID("web", nil),
+			entry:     basicEntry,
+			assertion: require.True,
+		},
+		{
+			name:    "wildcard name match",
+			service: NewServiceID("redis", nil),
+			entry: IngressGatewayConfigEntry{
+				Kind: "ingress-gateway",
+				Name: "ingress-web",
+				Listeners: []IngressListener{
+					{
+						Port:     1111,
+						Protocol: "http",
+						Services: []IngressService{
+							{
+								Name: "*",
+							},
+						},
+					},
+				},
+			},
+			assertion: require.True,
+		},
+		{
+			name:      "multiple listener match",
```

Comment:
Ah, the intention here is to test the fact that multiple listeners are defined on the config entry and we correctly iterate over the list. Not that `db` is declared in multiple listeners.

That being said, since converting to the memdb table, we don't use this function anymore, so I will remove it.